hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -228,11 +228,13 @@ public InFlightMetadataCollectorImpl(
 	private static SourceModelBuildingContext createModelBuildingContext(BootstrapContext bootstrapContext) {
 		final ClassLoaderService classLoaderService = bootstrapContext.getServiceRegistry().getService( ClassLoaderService.class );
 		final ClassLoaderServiceLoading classLoading = new ClassLoaderServiceLoading( classLoaderService );
-		return new SourceModelBuildingContextImpl(
-				classLoading,
-				bootstrapContext.getJandexView(),
-				ModelsHelper::preFillRegistries
-		);
+
+		final ModelsConfiguration modelsConfiguration = new ModelsConfiguration();
+		modelsConfiguration.setClassLoading(classLoading);
+//		modelsConfiguration.setExplicitContextProvider(  );
+		modelsConfiguration.configValue( "hibernate.models.jandex.index", bootstrapContext.getJandexView() );
+		modelsConfiguration.setRegistryPrimer( ModelsHelper::preFillRegistries );
+		return modelsConfiguration.bootstrap();
```

Comment:
Minor suggestion -

```suggestion
		return new ModelsConfiguration()
				.setClassLoading( classLoading )
				.setRegistryPrimer( ModelsHelper::preFillRegistries )
				.configValue( "hibernate.models.jandex.index", bootstrapContext.getJandexView() )
				.bootstrap();
```

Reply:
it seems this is not possible, the `setClassLoading` , `setRegistryPrimer` return `void` while `configValue` returns `Object` 
We have to change the `ModelsConfiguration` in `hibernate-models`

Reply:
Ok, let's ignore this then.  I misremembered.